src/codestyle.sh

```diff
@@ -73,6 +73,49 @@ function codestyle_main()
   checkpatch=$(join_path "$kernel_root" 'scripts/checkpatch.pl')
   cmd_script="perl ${checkpatch} ${checkpatch_options}"
 
+  [[ -n "${options_values['START_LINE']}" ]] && start_line=${options_values['START_LINE']}
+  [[ -n "${options_values['END_LINE']}" ]] && end_line=${options_values['END_LINE']}
+
+  if [[ -n "$start_line" || -n "$end_line" ]]; then
+    range_option_flag='true'
+  fi
+
+  # Check if --start-line and --end-line options are being used with a single file
+  if [[ ! -f "$path" && -n "$range_option_flag" ]]; then
+    complain "Invalid path using start-line and end-line option: ${path}"
+    return 2 # ENOENT
+  fi
+
+  # When no --start-line is specified, assign the first line of the file to start_line
+  if [[ -n "$range_option_flag" && -z "$start_line" ]]; then
+    start_line=1
+  fi
+
+  # When no --end-line is specified, assign the last line of the file to end_line
+  if [[ -n "$range_option_flag" && -z "$end_line" ]]; then
+    end_line=$(wc -l < "$FLIST")
+  fi
+
+  # If --start-line and --end-line options are being used, assign a temporary file
+  # to FLIST with the line interval and SPDX License Identifier
+  if [[ -n "$range_option_flag" ]]; then
+    suffix=".${FLIST#*.}"
+    temp_file=$(mktemp --suffix "$suffix" --tmpdir="$PWD")
+
+    if [[ "$start_line" != '1' ]]; then
+      printf '%s\n' '// SPDX-License-Identifier: TEMPFILE' > "$temp_file"
```

Comment:
Why do we need this?

Reply:
Every time we would run the command with --start-line greater than 1, checkpatch would give us the warning:

```
WARNING: Missing or malformed SPDX-License-Identifier tag in line 1
```

Even if the original file didn't have this issue. That's because a SPDX-License-Identifier is required in the first line of all files, if --start-line is greater than 1, the temporary file with the code snippet where we run checkpatch would not have this identifier, showing us this warning. 

To solve this issue we add this SPDX-License-Identifier to the first line of the temporary file every time --start-line is greater than 1. Even if the original file had this warning, it wouldn't make sense to show it in this case, because it's a warning that only happens in the first line.